bufstream/iceberg-quickstart/pkg/app/app.go

```diff
@@ -35,98 +48,183 @@ type Config struct {
 	CSR   csr.Config
 }
 
-// Main is used by the producer and consumer within their main functions.
+// Main is used by the consumer's main function.
 //
 // It sets up logging, interrupt handling, and binds and parses all flags. Afterwards, it calls
-// do to invoke the application logic.
-func Main(do func(context.Context, Config) error) {
+// action to invoke the application logic.
+func Main(action func(context.Context, Config) error) {
+	doMain(false, action)
+}
+
+// MainAutoCreateTopic is used by the producer's main function. It is just like [Main] except
```

Comment:
This is not the way to do this: `app.Main` is a top-level entry point that should do minimal setup work that is universal for all binaries. Instead, you've added a new `MainAutoCreateTopic` that does application logic. All application logic specific to the binary should be done within the `do` function that main invokes.